### PR TITLE
plugin/kubernetes: check for bare zone query

### PR DESCRIPTION
**Pull request by chrisohaver**
### 1. What does this pull request do?

checks for bare zone query and returns NODATA instead of all services.

### 2. Which issues (if any) are related?

#1687

### CODE REVIEW

File: plugin/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -273,6 +273,9 @@ func (k *Kubernetes) Records(state request.Request, exact bool) ([]msg.Service,
 	if e != nil {
 		return nil, e
 	}
+	if r.podOrSvc == ""{
```
Comment (bradbeam) [inline plugin/kubernetes/kubernetes.go]: nit probably want a space between `""` and `{`

**Review by bradbeam — CHANGES_REQUESTED** [review]

**Comment by chrisohaver** [comment]
FYI - I also found that bare `svc.cluster.local.` and `pod.cluster.local.` had bad behaviors.

`svc.cluster.local.` returned all services.
`pod.cluster.local.` returned `NXDOMAIN`

Both fixed in this PR now to return `NODATA`.

**Review by bradbeam — APPROVED** [review]
nice

File: plugin/kubernetes/parse.go [hunk 2]
```diff
@@ -35,6 +35,9 @@ func parseRequest(state request.Request) (r recordRequest, err error) {
 	// Federations are handled in the federation plugin. And aren't parsed here.
 
 	base, _ := dnsutil.TrimZone(state.Name(), state.Zone)
+	if base == "" || base == Svc || base == Pod {
```
Comment (miekg) [inline plugin/kubernetes/parse.go]: Small comment here? I.e. "return NODATA for apex queries?"

Comment (miekg) [reply plugin/kubernetes/parse.go]: apart from this nit: lgtm.

**Pull request merged**